src/Libraries/RevitServices/Events/EventHandlerProxy.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Autodesk.Revit.DB.Events;
+using Autodesk.Revit.UI.Events;
+using Dynamo.Models;
+
+namespace RevitServices.EventHandler
+{
+    /// <summary>
+    /// This is a event handler proxy class to serve as a proxy between the event publisher and
+    /// the event subscriber
+    /// </summary>
+    public class EventHandlerProxy
```

Comment:
Well, it is more like a service which depends only on Revit APIs and can be reused. What's more, a modularized implementation is better than putting everything in one place.

If deriving from one class and will cause issues, perhaps we need to make it not inheritable. But in this case, inheriting from this class won't cause any issues. Why not make it more open?